python/ray/data/tests/test_delta_sharing.py

```diff
@@ -0,0 +1,257 @@
+        }
+    ]
+    mock_datasource_instance._response.metadata = MagicMock()
+    mock_datasource_instance._response.metadata.schema_string = '{"type":"struct","fields":[{"name":"eventTime","type":"timestamp","nullable":true,"metadata":{}},{"name":"date","type":"date","nullable":true,"metadata":{}}]}'  # noqa E501
+    return mock_datasource
+
+
+@pytest.fixture
+def mock_to_pandas(mocker):
+    return mocker.patch("delta_sharing.reader.DeltaSharingReader._to_pandas")
+
+
+@pytest.fixture
+def mock_rest_client(mocker):
+    mock_rest_client = mocker.patch("delta_sharing.rest_client.DataSharingRestClient")
+    mock_rest_client_instance = mock_rest_client.return_value
+    mock_rest_client_instance.list_files_in_table.return_value = MagicMock(
+        add_files=[
+            {
+                "url": "https://s3-bucket-name.s3.us-west-2.amazonaws.com/delta-exchange-test/table2/date%3D2021-04-28/part-00000-591723a8-6a27-4240-a90e-57426f4736d2.c000.snappy.parquet",  # noqa E501
+                "id": "591723a8-6a27-4240-a90e-57426f4736d2",
+                "size": 573,
+                "partitionValues": {"date": "2021-04-28"},
+                "stats": '{"numRecords":1,"minValues":{"eventTime":"2021-04-28T23:33:48.719Z"},"maxValues":{"eventTime":"2021-04-28T23:33:48.719Z"},"nullCount":{"eventTime":0}}',  # noqa E501
+                "expirationTimestamp": 1652140800000,
+            }
+        ],
+        metadata=MagicMock(
+            schema_string='{"type":"struct","fields":[{"name":"eventTime","type":"timestamp","nullable":true,"metadata":{}},{"name":"date","type":"date","nullable":true,"metadata":{}}]}'  # noqa E501
+        ),
+    )
+    return mock_rest_client
+
+
+@pytest.fixture
+def setup_profile_file(tmpdir):
+    profile_content = {
+        "shareCredentialsVersion": 1,
+        "endpoint": "https://sharing.delta.io/delta-sharing/",
+        "bearerToken": "<token>",
+        "expirationTime": "2021-11-12T00:12:29.0Z",
+    }
+    profile_file = tmpdir.join("profile.json")
+    profile_file.write(json.dumps(profile_content))
+    return str(profile_file)
+
+
+def test_read_delta_sharing_tables(
+    mock_delta_sharing_datasource, mock_to_pandas, mock_rest_client, setup_profile_file
+):
+    # Mock the pandas DataFrame returned by _to_pandas
+    mock_to_pandas.return_value = MagicMock()
+
+    # Call the function with test parameters
+    dataset = read_delta_sharing_tables(
+        url=f"{setup_profile_file}#share.schema.table",
+        limit=10,
+        version=1,
+        timestamp="2021-11-12T00:12:29.0Z",
+    )
+
+    # Validate the results
+    mock_delta_sharing_datasource.assert_called_once_with(
+        url=f"{setup_profile_file}#share.schema.table",
+        jsonPredicateHints="{}",
+        limit=10,
+        version=1,
+        timestamp="2021-11-12T00:12:29.0Z",
+    )
+    assert mock_to_pandas.called
+    assert isinstance(dataset, MagicMock)
+
+
+if __name__ == "__main__":
+    unittest.main()
```

Comment:
We usually use pytest as the main function. 
Not sure if unittest also works well. 
But let's keep it consistent with other files.

```suggestion
if __name__ == "__main__":
    import sys

    sys.exit(pytest.main(["-v", __file__]))
```